addon/schema-manager.js

```diff
@@ -60,6 +61,10 @@ export class SchemaManager {
     return transform ? transform(value) : value;
   }
 
+  registerAsyncSchemas(schemas) {
```

Comment:
I'm not sure whether we want SchemaManager to be involved in asynchronous schemas - this is something we are coming up with and can be isolated in our specific implementation of the schema. The Schema Manager will always ask for fresh information irrespective.

Reply:
We do because nothing gets direct access to the schema, the manager is really just a singleton instance proxy

Reply:
In order to support dynamic schemas we probably need the schema to be a service, so we can throw it away between app instances. But the point I'm trying to make is this is not something, which the `SchemaManager` should be concerned with, IMHO. It is specific to the application and the particular usage of m3, how this is handled.

Reply:
The schema manager has the same API as the schema for a reason.. so that we can throw away the schema instance between test and application instances but maintain a stable API reference throughout the code..

Reply:
The schema manager is the internal API for interacting with the user's schema, but the schema object itself is userland.

The schema manager should have a minimal API (but it's fine for the userland schema to have a broader API).  For the schema manager to have an API that is not called internally is a smell.

It does not look to me that the schema manager needs to know about asynchronously loaded schemas.  What's the case where it does?

Reply:
@hjdivad so you're saying that in user-land registering an async-schema would be handled by an end-user directly manipulating `schema.models` for instance ?